x-pack/legacy/plugins/siem/public/pages/detection_engine/create_rule/components/step_define_rule/schema.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  // fieldFormatters,
+  fieldValidators,
+} from '../../../../../../../../../../src/plugins/es_ui_shared/static/forms/helpers';
+import { ERROR_CODE } from '../../../../../../../../../../src/plugins/es_ui_shared/static/forms/helpers/field_validators/types';
+
+import * as globalI18n from '../../translations';
+
+import { FieldValueQueryBar } from '../query_bar';
+import { CUSTOM_QUERY_REQUIRED, INVALID_CUSTOM_QUERY } from './translations';
+
+const { emptyField } = fieldValidators;
+
+export const schema: FormSchema = {
+  outputIndex: {
+    type: FIELD_TYPES.TEXT,
+    label: i18n.translate(
+      'xpack.siem.detectionEngine.createRule.stepDefineRule.fieldOutputIndiceNameLabel',
+      {
+        defaultMessage: 'Output index name',
+      }
+    ),
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate(
+            'xpack.siem.detectionEngine.createRule.stepDefineRule.outputIndiceNameFieldRequiredError',
+            {
+              defaultMessage: 'An output indice name for signals is required.',
+            }
+          )
+        ),
+      },
+    ],
+  },
+  useIndicesConfig: {
+    type: FIELD_TYPES.RADIO_GROUP,
+    label: i18n.translate(
+      'xpack.siem.detectionEngine.createRule.stepDefineRule.fieldIndicesTypeLabel',
+      {
+        defaultMessage: 'Indices type',
+      }
+    ),
+    options: [
+      {
+        id: 'true',
+        label: i18n.translate(
+          'xpack.siem.detectionEngine.createRule.stepDefineRule.indicesFromConfigDescription',
+          {
+            defaultMessage: 'Use Elasticsearch indices from SIEM advanced settings',
+          }
+        ),
+      },
+      {
+        id: 'false',
+        label: i18n.translate(
+          'xpack.siem.detectionEngine.createRule.stepDefineRule.indicesCustomDescription',
+          {
+            defaultMessage: 'Provide custom list of indices',
+          }
+        ),
+      },
+    ],
+  },
+  index: {
+    type: FIELD_TYPES.COMBO_BOX,
+    label: i18n.translate('xpack.siem.detectionEngine.createRule.stepAboutRule.fieldTagsLabel', {
+      defaultMessage: 'Indices',
+    }),
+    labelAppend: globalI18n.OPTIONAL_FIELD,
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate(
+            'xpack.siem.detectionEngine.createRule.stepDefineRule.outputIndiceNameFieldRequiredError',
+            {
+              defaultMessage: 'An output indice name for signals is required.',
+            }
+          )
+        ),
+      },
+    ],
+  },
+  queryBar: {
+    label: i18n.translate(
+      'xpack.siem.detectionEngine.createRule.stepDefineRule.fieldQuerBarLabel',
+      {
+        defaultMessage: 'Custom query',
+      }
+    ),
+    validations: [
+      {
+        validator: (
+          ...args: Parameters<ValidationFunc>
+        ): // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ReturnType<ValidationFunc<any, ERROR_CODE>> | undefined => {
```

Comment:
Optional: You can use this: `ReturnType<ValidationFunc<{}, ERROR_CODE>> | undefined => {`

and you won't need the any and linter disable